### PR TITLE
Resolved dialog context not found and added feature select tab and click command on dialog

### PR DESCRIPTION
**Pull request by atear10**
### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (updates to documentation, formatting, etc.)

### Description

Added a check for dialog on Select Tab and click command, if dialog is active click on dialog ribbon button or select tab on dialog else on main page. Also changed the xpath for dialog context as it is not unique to all environment

### Issues addressed

#1091 
#1092 

### All submissions:
- [x] My code follows the code style of this project.
- [ ] Do existing samples that are effected by this change still run?
- [ ] I have added samples for new functionality. 
- [ ] I raise detailed error messages when possible.
- [x] My code does not rely on labels that have the option to be hidden.

### Which browsers was this tested on?

- [x] Chrome
- [ ] Firefox
- [ ] IE
- [ ] Edge

### CODE REVIEW

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/Elements/Dialog.cs [hunk 1]
```diff
@@ -190,5 +190,28 @@ public bool SetStateDialog(bool clickOkButton)
...
+        /// <param name="subButtonName"></param>
+        /// <param name="secondSubButtonName"></param>
+        /// <returns></returns>
+        public bool ClickCommand(string buttonName, string subButtonName = null, string secondSubButtonName = null)
+        {
+            return _client.ClickCommand(buttonName, subButtonName, secondSubButtonName);
+        }
+
+        /// <summary>
+        /// Clicks on entity dialog ribbon button
```
Comment (TYLEROL) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/Elements/Dialog.cs:207]: nit: Can we update the summary info here

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/DTO/AppElementReference.cs [hunk 2]
```diff
@@ -418,7 +418,7 @@ public static class AppElements
             { "Entity_RecordSetNavList", "//ul[contains(@data-id, 'recordSetNavList')]" },
             { "Entity_RecordSetNavCollapseIcon", "//*[contains(@data-id, 'recordSetNavCollapseIcon')]" },
             { "Entity_RecordSetNavCollapseIconParent", "//*[contains(@data-id, 'recordSetNavCollapseIcon')]" },
-            { "Entity_TabList", "//ul[contains(@id, \"tablist\")]" },
+            { "Entity_TabList", ".//ul[contains(@id, \"tablist\")]" },
```
Comment (TYLEROL) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/DTO/AppElementReference.cs:421]: Can you confirm that Entity.SelectTab() still functions with this change?

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/DTO/AppElementReference.cs [hunk 3]
```diff
@@ -492,7 +492,7 @@ public static class AppElements
             { "Entity_Header_DateTimeFieldContainer","//div[@data-id='header_[NAME]-FieldSectionItemContainer']" },
                         
             //CommandBar
-            { "Cmd_Container"       , "//ul[contains(@data-lp-id,\"commandbar-Form\")]"},
+            { "Cmd_Container"       , ".//ul[contains(@data-lp-id,\"commandbar-Form\")]"},
```
Comment (TYLEROL) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/DTO/AppElementReference.cs:495]: Can you confirm that the following methods still function with this change?

- CommandBar.GetCommandValues();
- CommandBar.ClickCommand();

**Comment by TYLEROL** [comment]
@atear10 - Could you refresh this branch of yours with the latest from develop? It looks like you have some commandBar oriented changes that haven't sync'd from changes merged to develop yesterday.

Also, there are a couple of of xpath changes with using the '.' expression that I want to make sure don't impact existing methods. Comments included on the PR to confirm.

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/DTO/AppElementReference.cs (hunk 2 above)
Comment (atear10) [reply Microsoft.Dynamics365.UIAutomation.Api.UCI/DTO/AppElementReference.cs:421]: Yes with this change it works well with both main page tab and dialog tab , i have already tested this

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/DTO/AppElementReference.cs (hunk 3 above)
Comment (atear10) [reply Microsoft.Dynamics365.UIAutomation.Api.UCI/DTO/AppElementReference.cs:495]: yes this also works I am already using this at both places

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/Elements/Dialog.cs (hunk 1 above)
Comment (atear10) [reply Microsoft.Dynamics365.UIAutomation.Api.UCI/Elements/Dialog.cs:207]: done

**Review by TYLEROL — APPROVED** [review]

**Comment by atear10** [comment]
@TYLEROL when we are releasing the latest bits nuget as this is really needed now

**Pull request merged**